### DISCUSSION TITLE
Problems with Amazon WorkSpaces and Cisco Webex

### DISCUSSION THREAD

**Discussion by geofftench** [ttscoff/bunch-docs#119] [Q&A] [answered]
Hi,

I have just about the simplest possible bunch, and I have problems with two out of the three referenced applications:

```
Webex
Slack
WorkSpaces
```

**Slack** works fine, so I won't mention it again.

**WorkSpaces** is [Amazon WorkSpaces client](https://clients.amazonworkspaces.com). It reliably starts when I launch the bunch, but never quits, with the following error being written into the log:

`App not running (WorkSpaces)`

**Webex** is the [Webex client](https://www.webex.com/downloads.html). It starts roughly half the time, but if it started, it will reliably quit. Nothing different is written into the log whether the application launches or not.

Both applications have names that match the `kMDItemDisplayName` tag.

Any tips on how to diagnose the problems?

Thanks very much.

**Comment by ttscoff** [comment]
I don't have either of these apps to test, but did you go through the 
troubleshooting section? 
https://bunchapp.co/docs/troubleshooting/#appnames

You probably need to find out whether the apps change names once they're 
running, the way that Logic Pro X, Visual Studio Code and some others 
do. Do the apps respond to `tell app X to launch` and `tell app X to 
quit` in Script editor, where X is the same both times? When WorkSpaces 
is running, is there a process in `ps -ax` that has a different name? 
Most of this is covered in the troubleshooting section, let me know if 
it helps.

-Brett

**Comment by ttscoff** [comment, marked as answer]
I downloaded Webex to try it out and it seems to launch/quit 
consistently for me, which makes it hard for me to help diagnose that 
one. I'll try WorkSpaces too and see if I can at least reproduce that 
one.

What OS are you on, and are your running an Intel or M1 mac?

-Brett

**Reply by ttscoff** [reply]
Ok, yeah, at least on Big Sur on an M1 Mac, WorkSpaces launches but doesn't quit for me, with the same log error you showed. It does, however, quit from Script Editor with `tell app "WorkSpaces" to quit`. It shows up in the process list as `WorkSpaces`, so that's not helpful either.

The solution to that one is going to be manually running the AppleScript to quit the app:

```
%WorkSpaces
!* tell app "WorkSpaces" to quit
```

The % will avoid Bunch trying to quit it itself, and the `!*` will run the AppleScript when closing the Bunch.

Still unable to reproduce the Webex issue, though.

**Reply by geofftench** [reply]
Thanks Brett, that's really helpful. I'll see what additional information I can collect on the Webex issue I'm having - it's probably going to turn out to be something odd that's specific to my machine.

**Reply by geofftench** [reply]
One additional point that I can add -- if I go trawling through the process tree, when the **Webex** application doesn't start, I see the following entry only:

```
/Users/<username>/Library/Application Support/WebEx Folder/Add-ons/Cisco WebEx Start.app/Contents/MacOS/Cisco WebEx Start -daemon 1 -LaunchedBy /Users/<username>/Library/Application Support/WebEx Folder/Add-ons/Cisco WebEx Start.app -JoinHistory transurban.webex.com
```

When it starts successfully, I see the following processes:

```
/Applications/Webex.app/Contents/MacOS/Webex Teams

/Users/<username>/Library/Application Support/Cisco Spark/Webexteams_upgrades/41.7.0.19440_60929bbc-1903-408f-b3f6-775c54e80eda/Webex Teams.app/Contents/PlugIns/CiscoSparkPlugin.bundle/Contents/MacOS/WebexHelper /ipcId=22270_1 /processId=22270 /hangDetectionTimeout=30 /bundleNameAndPath=/Users/<username>/Library/Application Support/Cisco Spark/Webexteams_upgrades/41.7.0.19440_60929bbc-1903-408f-b3f6-775c54e80eda/Webex Teams.app/Contents/PlugIns/CiscoSparkPlugin.bundle/Contents/PlugIns/CiscoSparkMedia.bundle /clientVersion=41.7.0.19440

/Users/<username>/Library/Application Support/Cisco Spark/Webexteams_upgrades/41.7.0.19440_60929bbc-1903-408f-b3f6-775c54e80eda/Webex Teams.app/Contents/PlugIns/CiscoSparkPlugin.bundle/Contents/Helpers/CaptureHost.app/Contents/MacOS/CaptureHost /communicationChannelId=22284_2 /defaultChannelId=22284_1

/Users/<username>/Library/Application Support/Cisco Spark/Webexteams_upgrades/41.7.0.19440_60929bbc-1903-408f-b3f6-775c54e80eda/Webex Teams.app/Contents/PlugIns/CiscoSparkPlugin.bundle/Contents/MacOS/WebexHelper /ipcId=22270_2 /launchCount=1 /processId=22270 /bundleNameAndPath=/Users/<username>/Library/Application Support/Cisco Spark/Webexteams_upgrades/41.7.0.19440_60929bbc-1903-408f-b3f6-775c54e80eda/Webex Teams.app/Contents/PlugIns/CiscoSparkPlugin.bundle/Contents/PlugIns/CiscoSparkMercury.bundle /instanceName=#1/MainConnection/#0/Process
```

Perhaps I have a stale application hanging around under `Application Support` from an earlier version of the Webex client... I'm not sure what causes one to be chosen over the other, if that's what's happening, though.

**Reply by geofftench** [reply]
Deleting `~/Library/Application Support/WebEx Folder` seems to have resolved the **Webex** problem, and it now starts and stops reliably. Thanks again for your help :)

**Comment by ttscoff** [comment]
It will probably recreate the support folder, so keep me posted if this 
starts happening again.

Thanks,
Brett

**Reply by geofftench** [reply]
You were exactly right, it recreated the folder, and then the issue started again. I'm trying the bundle identifier (`Cisco-Systems.Spark`) in the Bunch file instead, will see whether that makes a difference.

**Reply by geofftench** [reply]
This combination seems to work... so far... (I found that **Webex** didn't always quit when using the bundle name.)

```
Slack
%Cisco-Systems.Spark
!* tell app "Webex" to quit
%WorkSpaces
!* tell app "WorkSpaces" to quit
```

**Reply by ttscoff** [reply]
Well, that's a lot more work than it's supposed to be, but if that's
working for now, great. Keep me posted, if this continues working I'll
add it as a tip in the troubleshooting docs.

**Reply by geofftench** [reply]
I guess I'm just lucky, having badly behaved apps as part of my day job. This seems to be doing the trick so far.
